### PR TITLE
feat: add proxy support

### PR DESCRIPTION
**Pull request by QuentinN42**
Only the AWS is not currently implemented with proxy.
There is a log line to ask users to contribute if they want this feature.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -31,10 +31,9 @@ jobs:
         if: always()
         run: |
           source .venv/bin/activate
-          isort -m 9 --line-length 159 $MODULE_NAME tests --check-only
-          pylint --load-plugins pylint_quotes $MODULE_NAME tests
-          yapf -rd $MODULE_NAME tests
+          ruff -V
           mypy -V
+          ruff check $MODULE_NAME tests
```
Comment (c3b5aw) [inline .github/workflows/ci.yaml:36]: Well ruff is not a dependency both in pyproject and as an extension.

If you want to pass this change you must also remove these deprecated dependencies from pyproject.

**Review by c3b5aw — CHANGES_REQUESTED** [review]
Looks good, will do further checking.

What's missing for aws?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (c3b5aw) [reply .github/workflows/ci.yaml:36]: Looks like linting has been ninja'ed already.

https://github.com/Escape-Technologies/py-multiauth/commit/f0947f30a476fa8ee500c9216606b31120513d07

Comment (QuentinN42) [reply .github/workflows/ci.yaml:36]: Done

**Review by c3b5aw — APPROVED** [review]

**Pull request merged**